src/lib/components/CommunitySelectionModal/CommunitySelectionModal.js

```diff
@@ -7,35 +7,35 @@
 import { i18next } from "@translations/i18next";
 import PropTypes from "prop-types";
 import React, { Component } from "react";
+import { connect } from "react-redux";
 import { Header, Modal } from "semantic-ui-react";
 import { CommunityContext } from "./CommunityContext";
 import { CommunitySelectionSearch } from "./CommunitySelectionSearch";
 
-export class CommunitySelectionModal extends Component {
+class CommunitySelectionModalComponent extends Component {
   constructor(props) {
     super(props);
-    const { chosenCommunity, onCommunityChange } = props;
+    const { chosenCommunity, onCommunityChange, userCommunitiesMemberships } = props;
 
     this.state = {
       modalOpen: false,
       localChosenCommunity: chosenCommunity,
     };
 
-    const { localChosenCommunity } = this.state;
-
     this.contextValue = {
       setLocalCommunity: (community) => {
         onCommunityChange(community);
         this.setState({ localChosenCommunity: null, modalOpen: false });
       },
-      getChosenCommunity: () => localChosenCommunity,
+      // eslint-disable-next-line react/destructuring-assignment
+      getChosenCommunity: () => this.state.localChosenCommunity,
```

Comment:
nit: why not destructuring assignment?

Reply:
This is intended as the function otherwise leads to a bug where the state.chosenCommunity is never updated! Maybe a comment on why is a good idea so we don’t introduce the bug again!